src/LinuxTracing/LinuxTracingIntegrationTest.cpp

```diff
@@ -946,24 +960,34 @@ TEST(LinuxTracingIntegrationTest, ThreadNames) {
 
   VerifyOrderOfAllEvents(events);
 
-  std::vector<std::string> collected_event_names;
+  std::vector<std::string> changed_thread_names;
+  std::vector<std::string> initial_thread_names;
   for (const auto& event : events) {
+    if (event.event_case() == orbit_grpc_protos::ProducerCaptureEvent::kCaptureStarted) {
+      const orbit_grpc_protos::CaptureStarted& capture_started = event.capture_started();
+      for (const auto& thread_name : capture_started.thread_names()) {
+        if (thread_name.pid() != fixture.GetPuppetPid()) {
+          continue;
+        }
+
+        initial_thread_names.emplace_back(thread_name.name());
+      }
+    }
     if (event.event_case() != orbit_grpc_protos::ProducerCaptureEvent::kThreadName) {
       continue;
     }
 
     const orbit_grpc_protos::ThreadName& thread_name = event.thread_name();
-    if (thread_name.tid() != fixture.GetPuppetPid()) {
+    if (thread_name.pid() != fixture.GetPuppetPid()) {
       continue;
     }
 
-    EXPECT_EQ(thread_name.pid(), fixture.GetPuppetPid());
```

Comment:
Why is this being changed? The point here was to filter based on the `tid` and then verify that the `pid` is equal to the `tid`. This is because the puppet is single threaded. You are basically removing the test on `tid == pid` like this.

Reply:
if tid != puppet_pid, this one is always true, I restore the check in but now it is checking for tid.